core-bundle/src/Command/MigrateCommand.php

```diff
@@ -362,6 +364,12 @@ private function executeMigrations(bool &$dryRun, bool $asJson, string $specifie
                 // Do not run the update recursive if a hash was specified
                 break;
             }
+
+            if ($loopControl-- < 1) {
+                $this->io->error('The migrations were stopped after 9 iterations as a precaution. There is a high chance of an infinite loop of migrations. If this is not the case, please re-run the command. To troubleshoot this error, check the `shouldRun()` method of the migration(s) listed above.');
```

Comment:
We also need to handle `$asJson` here.
Either by using `$this->writeNdjson('error', […` or by throwing an exception (which gets converted to NDJSON automatically).